lib/Storage/MongoDbTimeSeriesStore.php

```diff
@@ -533,46 +557,27 @@ public function getStats($filterBy = [])
                         'widgetId'=> 1,
                         'mediaId'=> 1,
                         'statDate'=> 1,
+                        'engagements'=> 1,
                     ]
                 ],
             ];
 
-            // Sort by id (statId) - we must sort before we do pagination as mongo stat has descending order indexing on start/end
-            $query[]['$sort'] = ['id'=> 1];
-
             if ($start !== null && $length !== null) {
                 $query[]['$skip'] =  $start;
                 $query[]['$limit'] = $length;
             }
 
             $cursor = $collection->aggregate($query);
 
-        } catch (\MongoDB\Exception\RuntimeException $e) {
-            $this->log->error($e->getMessage());
-        }
+            $result = new TimeSeriesMongoDbResults($cursor);
 
-        $result = new TimeSeriesMongoDbResults($cursor);
-
-        // Get total
-        try {
-            $totalQuery = [
-                $match,
-                [
-                    '$group' => [
-                        '_id'=> null,
-                        'count' => ['$sum' => 1],
-                    ]
-                ],
-            ];
-            $totalCursor = $collection->aggregate($totalQuery);
+            // Total
+            $result->totalCount = $total;
 
         } catch (\MongoDB\Exception\RuntimeException $e) {
             $this->log->error($e->getMessage());
+            throw new GeneralException($e->getMessage());
```

Comment:
Do we really want to pass e->getMessage out to the user? Will that mean anything to them at this point?

We could pass them a user friendly message instead?

```
 throw new GeneralException(__('Sorry we encountered an error getting Proof of Play data, please consult your administrator'));
```